docs/data/data-grid/column-definition/column-definition.md

```diff
@@ -130,6 +130,47 @@ You can check the [styling cells](/x/react-data-grid/style/#styling-cells) secti
 Cell content should not be in the tab sequence except if cell is focused.
 You can check the [tab sequence](/x/react-data-grid/accessibility/#tab-sequence) section for more information.
 
+### Using hooks inside a renderer
+
+The `renderCell` property is a function returning a React node, not a React component.
+
+If you want to use React hooks inside your renderer, you should wrap them inside a component.
+
+```tsx
+// ❌ Not valid
```

Comment:
I'd be interested to know more about why this doesn't work.

Reply:
React relies on the call order of the hooks to retain information across renders
So the hooks executed must always be the same across renders (no hook in a `if` branch for instance).

If you put a hook directly in the render function, depending on whether or not the function is called, the component calling it will have a different number of hooks executed.
Which causes the error.

That's a very basic React topic, but hidden inside a render function so easy to miss.